main.go

```diff
@@ -56,6 +63,20 @@ func realMain() int {
 		}
 	}()
 
+	// Create the version checker
+	checker, err := versioncheck.New(io, versionCheckStatePath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to configure version checker: %v\n", err)
+		return 1
```

Comment:
Should this just be a warning and not cause the application to exit? 

Especially if we expect `New` to create the folders it would otherwise prevent the CLI from being used without write access.